config/routes.rb

```diff
@@ -2,8 +2,10 @@
 
   root 'pages#show', page:'home'
 
-  resources :lessons
-  get '/lessons/publish/:id' => 'lessons#publish'
+  resources :lessons do
+    get 'publish', on: :member
```

Comment:
This `on: :member` doesn't make much sense, should just be `get 'publish'`. Also this should break, since publish now uses the `find_lesson` action, which checks for a lesson based on `params[:id]`, while this should now use `params[:lesson_id]` based on the documentation.

Also would suggest moving comment inside this do block.